tests/cache/test_mesi_controller.py

```diff
@@ -227,14 +261,21 @@ def test_processor_write_exclusive(self):
         address = 0x1000
 
         # Call the function under test
-        next_state = self.controller.handle_processor_request(MESIState.EXCLUSIVE, address, is_processor_write=True)
+        next_state = self.controller.handle_processor_request(
+            MESIState.EXCLUSIVE, address, is_processor_write=True
+        )
 
         # Check that the bus operation was not called, silent upgrade
         self.mock_bus_op.assert_not_called()
 
         # Check that the snoop result was not checked
         self.mock_get_snoop.assert_not_called()
 
+        # Check that sent line was not called
+        # L1 is write through on first write, so L1 and L2 will have modified line at the
+        # same time
+        self.mock_l1_message.assert_not_called()
```

Comment:
I would've thought this state would also need to send the line to L1, because it means L1 doesn't have the line but wants to write a byte to it. But I think what I'm really getting confused about is how we know if L1 is writing back due to a local eviction vs writing in order to simply write more data to the line. Because in one case, we should send the line back to L1, but in the other, we should not. Does that make sense? I'm not sure if I'm missing something obvious, but this suddenly seems very unclear to me.

Reply:
ugh, you are right. I fixed it up. Hopefully, everything is covered now.